checkov/secrets/runner.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import os
+import re
+from typing import Optional
+
+from detect_secrets.core.potential_secret import PotentialSecret
+from detect_secrets.core.scan import scan_file
+from detect_secrets.core.usage import initialize_plugin_settings
+
+from checkov.common.comment.enum import COMMENT_REGEX
+from checkov.common.models.enums import CheckResult
+from checkov.common.output.record import Record
+from checkov.common.output.report import Report
+from checkov.common.runners.base_runner import BaseRunner, filter_ignored_directories
+from checkov.runner_filter import RunnerFilter
+import linecache
+
+
+SECRET_TYPE_TO_ID = {
+    'Artifactory Credentials': 'CKV_SECRET_1',
+    'AWS Access Key': 'CKV_SECRET_2',
+    'Azure Storage Account access key': 'CKV_SECRET_3',
+    'Basic Auth Credentials': 'CKV_SECRET_4',
+    'Cloudant Credentials': 'CKV_SECRET_5',
+    'Base64 High Entropy String': 'CKV_SECRET_6',
+    'IBM Cloud IAM Key': 'CKV_SECRET_7',
+    'IBM COS HMAC Credentials': 'CKV_SECRET_8',
+    'JSON Web Token': 'CKV_SECRET_9',
+    'Secret Keyword': 'CKV_SECRET_10',
+    'Mailchimp Access Key': 'CKV_SECRET_11',
+    'NPM tokens': 'CKV_SECRET_12',
+    'Private Key': 'CKV_SECRET_13',
+    'Slack Token': 'CKV_SECRET_14',
+    'SoftLayer Credentials': 'CKV_SECRET_15',
+    'Square OAuth Secret': 'CKV_SECRET_16',
+    'Stripe Access Key': 'CKV_SECRET_17',
+    'Twilio API Key': 'CKV_SECRET_18',
+    'Hex High Entropy String': 'CKV_SECRET_19'
+}
+
+PROHIBITED_FILES = ['Pipfile.lock', 'yarn.lock', 'package-lock.json', 'requirements.txt']
+
+
+class Runner(BaseRunner):
+    check_type = 'secrets'
+
+    def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(),
+            collect_skip_comments=True) -> Report:
+        report = Report(self.check_type)
+        initialize_plugin_settings(None)
+        # Implement non IaC files (including .terraform dir)
+        files_to_scan = files or []
+        excluded_paths = (runner_filter.excluded_paths or []) + ['node_modules', '.terraform']
```

Comment:
@tronxd Can `['node_modules', '.terraform']` be global? (maybe used like in base runner `IGNORED_DIRECTORIES_ENV = os.getenv('CKV_IGNORED_DIRECTORIES', "node_modules,.terraform,.serverless")`)

Reply:
Sure